Automattic/sensei
# Display lesson action blocks in the frontend

### Changes proposed in this Pull Request

* This PR handles frontend rendering of the lesson actions block.
* The old template deprecation will happen as part of another PR so the old buttons/notices are still displayed.
* The rules of when each button will appear are the following:
  * The 'Next Lesson' button will appear when the course is completed.
  * The 'Reset Lesson' button will appear when the user has completed the course, there is a quiz linked to it and the quiz reset button is enabled in both block and quiz settings.
  * The 'Complete Lesson' button will appear when the user has not completed the course and there is no required quiz to the lesson.
  * The 'View Quiz' button will appear when there is a quiz linked to the course and the lesson is not completed.

### Testing instructions

* For the described cases, observe that the buttons are displayed correctly in the frontend.
* Try various settings on the blocks and observe that they are displayed correctly in the frontend.
